src/InfRcTransport.cc

```diff
@@ -861,7 +880,7 @@ InfRcTransport::ServerConnectHandler::handleFileEvent(int events)
     // complete the initialisation.
     QueuePairTuple outgoingQpt(downCast<uint16_t>(transport->lid),
                                qp->getLocalQpNumber(),
-                               qp->getInitialPsn(), incomingQpt.getNonce());
+                               qp->getInitialPsn(), incomingQpt.getNonce(), gid);
```

Comment:
Can we make a command-line guard for RoCE and use it anywhere we attempt something for RoCE that could break for non-RoCE communication?